crawler.py

```
@@ -1,45 +1,70 @@
 import mechanize
-import urllib.parse
+from collections import namedtuple
+from urllib.parse import urljoin, urlparse
 
-
-def is_not_absolute(url):
-    # Check if url is an absolute or not
-    if '//' not in url:
+def validated(url):
+    '''Checks if given url is proper.'''
+    browser = mechanize.Browser()
+    browser.set_handle_robots(False)
+    try:
+        browser.open(url)
         return True
-    else:
+    except:
```

Using `except` may mask all other exceptions that you may stumble upon. If you are expecting any please be more explicit.

This is an example what happens when you do not specify exception. This should raise `NameError`, and it will if I add `KeyError` to `except`.
```
try:
    print(d['id'])
except:
    print('No such id')
```